openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -198,13 +201,23 @@ def celebrations(self):
         Returns a list of celebrations that should be performed.
         """
         browser_timezone = self.request.query_params.get('browser_timezone', None)
-        return {
+        streak_length_to_celebrate = UserCelebration.perform_streak_updates(
+            self.effective_user, self.course_key, browser_timezone
+        )
+        celebrations = {
             'first_section': CourseEnrollmentCelebration.should_celebrate_first_section(self.enrollment_object),
-            'streak_length_to_celebrate': UserCelebration.perform_streak_updates(
-                self.effective_user, self.course_key, browser_timezone
-            ),
+            'streak_length_to_celebrate': streak_length_to_celebrate,
+            'streak_discount_experiment_enabled': False,
         }
 
+        # We only want to bucket people into the AA-759 experiment if they are going to see the streak celebration
+        if streak_length_to_celebrate:
+            # We only want to bucket people into the AA-759 experiment
+            # if the course has not ended, is upgradeable and the user is not an enterprise learner
+            if AA759_can_show_coupon(self.effective_user, self.course):
+                celebrations['streak_discount_experiment_enabled'] = AA759_FLAG.is_enabled()
```

Comment:
Good job saving this call to the last bit you could - since calling this will emit the bucketing segment signal.

Reply:
yup, that's why I did it this way (although I did think it working this way could easily lead to a bug for someone not familiar with that behavior)
I wonder if there could be some future improvement with a query parameter or something that makes it clear there is a side effect happening

Reply:
The method `get_bucket()` has a documented `track=True` parameter that controls the side effect. It's possible that should also be added to the `is_enabled` method, to help visiblity.

The side effect is also documented on the wiki page for this flag class. But yeah, overall, easy to miss if you're just consuming the class and not doing a book report on it.